unstructured_ingest/v2/processes/connectors/couchbase.py

```diff
@@ -0,0 +1,157 @@
+        sensitive=True, default_factory=CouchbaseAccessConfig
+    )
+    bucket: Optional[str] = None
+    scope: Optional[str] = None
+    collection: Optional[str] = None
+    batch_size: int = 50
+    connector_type: str = CONNECTOR_TYPE
+
+
+@dataclass
+class CouchbaseUploadStagerConfig(UploadStagerConfig):
+    pass
+
+
+@dataclass
+class CouchbaseUploadStager(UploadStager):
+    upload_stager_config: CouchbaseUploadStagerConfig = field(
+        default_factory=lambda: CouchbaseUploadStagerConfig()
+    )
+
+    def run(
+        self,
+        elements_filepath: Path,
+        file_data: FileData,
+        output_dir: Path,
+        output_filename: str,
+        **kwargs: Any,
+    ) -> Path:
+        with open(elements_filepath) as elements_file:
+            elements_contents = json.load(elements_file)
+
+        output_path = Path(output_dir) / Path(f"{output_filename}.json")
+        with open(output_path, "w") as output_file:
+            json.dump(elements_contents, output_file)
+        return output_path
+
+
+@dataclass
+class CouchbaseUploaderConfig(UploaderConfig):
+    batch_size: int = 50
+
+
+@dataclass
+class CouchbaseUploader(Uploader):
+    upload_config: CouchbaseUploaderConfig
+    connection_config: CouchbaseConnectionConfig
+    cluster: Optional["Cluster"] = field(init=False, default=None)
+    connector_type: str = CONNECTOR_TYPE
+
+    def __post_init__(self):
+        self.cluster = self.connect_to_couchbase()
+
+    @requires_dependencies(["couchbase"], extras="couchbase")
+    def connect_to_couchbase(self) -> "Cluster":
+        from datetime import timedelta
+
+        from couchbase.auth import PasswordAuthenticator
+        from couchbase.cluster import Cluster
+        from couchbase.options import ClusterOptions
+
+        connection_string = username = password = None
+        access_conf = self.connection_config.access_config
+        try:
+            if access_conf.connection_string is not None:
+                connection_string = access_conf.connection_string
+            if access_conf.username is not None:
+                username = access_conf.username
+            if access_conf.password is not None:
+                password = access_conf.password
+        except Exception as e:
+            raise f"please provide connection string, username and password : {e}"
+
+        auth = PasswordAuthenticator(username, password)
+        options = ClusterOptions(auth)
+        options.apply_profile("wan_development")
+        cluster = Cluster(connection_string, options)
+        cluster.wait_until_ready(timedelta(seconds=5))
+        return cluster
+
+    def run(self, contents: list[UploadContent], **kwargs: Any) -> None:
+        elements_dict = []
+        for content in contents:
+            with open(content.path) as elements_file:
+                elements = json.load(elements_file)
+                # Modify the elements to match the expected couchbase format
```

Comment:
This sort of manipulation should live in the stager. The uploader itself shouldn't be responsible for modifying the data by the time it gets it.

Reply:
Added the element manipulation logic in stager